src/Esprima/JavascriptParser.cs

```diff
@@ -76,12 +74,16 @@ public void ReleaseLargeBuffers()
         public ArrayList<Decorator> Decorators;
 
         public HashSet<string?> LabelSet;
+
+        // we use an boxed object instead of Token to reduce stack size and improve
+        // *CoverGrammar method inlining, this is used only for error conditions
+        public object? FirstCoverInitializedNameError;
```

Comment:
nit: Could use [`StrongBox<Token>?`](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.strongbox-1?view=netframework-4.6.2) here [to avoid casting](https://github.com/dotnet/runtime/discussions/47774).

Reply:
Let's try that one out!